src/algorithms/calorimetry/CalorimeterHitReco.cc

```diff
@@ -139,19 +140,23 @@ void CalorimeterHitReco::AlgorithmProcess() {
 //        #pragma GCC diagnostic error "-Wsign-converstion"
 
         //did not pass the zero-suppresion threshold
+        const auto cellID = rh->getCellID();
         if (rh->getAmplitude() < m_pedMeanADC + thresholdADC) {
             continue;
         }
 
         // convert ADC to energy
-        const float energy =
-                (((signed) rh->getAmplitude() - (signed) m_pedMeanADC)) / static_cast<float>(m_capADC) * m_dyRangeADC /
+        float energy = (((signed) rh->getAmplitude() - (signed) m_pedMeanADC)) / static_cast<float>(m_capADC) * m_dyRangeADC /
                 m_sampFrac;
-        const float time = rh->getTimeStamp() / stepTDC;
+        if (m_readout == "LFHCALHits" && m_sampFracLayer[0] != 0.){
+          energy = (((signed) rh->getAmplitude() - (signed) m_pedMeanADC)) / static_cast<float>(m_capADC) * m_dyRangeADC /
+                    m_sampFracLayer[decoder->get(cellID, 7)];
```

Comment:
What is "7" here?

Reply:
`   
  <readout name="LFHCALHits">
       <segmentation type="NoSegmentation"/> -->
      <id>system:8,moduleIDx:6,moduleIDy:6,moduletype:1,passive:1,towerx:2,towery:1,rlayerz:4,layerz:4</id>
    </readout>
`
its the readout layer index (rlayerz)

Reply:
How about we replace `7` with `id_dec->index(m_layerField)`?